### DISCUSSION TITLE
Is there a way to include style using a dynamic assembly name ?

### DISCUSSION THREAD

**Discussion by Ehrode** [AvaloniaUI/Avalonia#13899] [Q&A]
Lets say I have my styles shared by multiple projects inside Foo.MyStyles.

To include a style I'll need to add :
`  <StyleInclude Source="avares://Foo.MyStyles/styles_files.axaml"/> 
`In my target project.
Now the issue is that "Foo.MuStyles" has a dynamic assembly names depending of my configuration.
When I build the generate dll could be Foo.MyStyles.dll but also AnotherName.MyStyles.Dll.
The StyleInclude stops working instantly and I can't build anymore the moment I change my Styles project assembly name.

I managed to get it working using a class that inherit styles and import it from a using, but when I do it like this my IDE stops seeing that the resources are affectively imported and I end up with a warning "class not found" or "resource not found" on all my references to the style project which isn't an option.

I just blocked my dynamic assembly name for now but we are developing an app with a "white label" feature and our dll must have a name that changes according to the client which is not possible at the moment.

Is there any solution I couldn't think of or this is just not possible at the moment?

**Comment by timunie** [comment]
You can also load the styles in C# if that helps you. You are not forced to use XAML.

**Reply by Ehrode** [reply]
If you mean from app.axaml.cs adding Styles.Add(new StyleInclude(){Uri=new uri("avares://....")}) I could do what I want like this.
But when you do it like this, the previewer stop seeing your classes as their were not included (it works at runtime but in design time I have resource not found on all my style classes).
This is annoying because I can't use the IDE anymore to see resources that really doesn't exists and has issue in import, they are all flagged as not found.

**Reply by timunie** [reply]
Idea: Add a base set of styles as before in XAML. At run-time change them. Or check out what FluentTheme.cs does. https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/FluentTheme.xaml.cs

**Comment by maxkatz6** [comment]
Keep in mind, while you can do it with dynamic names in C#, you code won't be trimming/AOT friendly anymore, and these XAML files can be removed by the compiled. Which means, you would have to disable trimming.
You can, alternatively, define these XAML files with x:Class (and add a switch over them in C#) or include all files in XAML at once, and remove and manipulate over them in C#. Or change the way how your app works here as a third option.

**Reply by Ehrode** [reply]
"You can, alternatively, define these XAML files with x:Class (and add a switch over them in C#) or include all files in XAML at once"

Can you detail a bit how to acheive this?

What I did so far is creating a styles axaml file in my shared style library and a class behind (with x:Class). Then in the app.axaml of my project I defined xmlns:foo="using:MySharedStyleProject" then in the App.Styles section <foo:MyClassWithAllTheStylesIncluded/>

This works as my app works with all the styles correctly applied, but the issue with this is that the IDE (Rider) warn me that all my styles are not found during design time.

![image](https://github.com/AvaloniaUI/Avalonia/assets/40527875/91ddd2ea-cb3a-4a68-9814-cf3a5921b573)

This doesn't prevent the app to works correctly but I loose the ability to see when I'm using a class that is not correctly included or named and have missing styles that I won't be able to see until I spot it while the app is running.
